app/src/organisms/ProtocolUpload/hooks/useCurrentProtocol.ts

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react'
+import last from 'lodash/last'
+import { useProtocolQuery } from '@opentrons/react-api-client'
+import { useCurrentRun } from './useCurrentRun'
+
+import type { Protocol } from '@opentrons/api-client'
+
+export function useCurrentProtocol(): Protocol | null {
+  const currentRun = useCurrentRun()
+
+  const enableProtocolPolling = React.useRef<boolean>(
+    currentRun?.data.protocolId != null
+  )
```

Comment:
I think this initial state is unnecessary and it can just start at `true`. If `currentRun?.data.protocolId` is undefined,  `useProtocolQuery` will disable itself; we don't need to tell it polling is disabled, too

```suggestion
  const enableProtocolPolling = React.useRef(true)
```